api/src/exam-environment/routes/exam-environment.ts

```diff
@@ -1000,22 +1000,21 @@ export async function getExamAttemptsByExamIdHandler(
 /**
  * Gets all the relations for a given challenge and exam(s).
  */
-export async function getExamMappingsByChallengeId(
+export async function getExamChallenge(
   this: FastifyInstance,
-  req: UpdateReqType<
-    typeof schemas.examEnvironmentGetExamMappingsByChallengeId
-  >,
+  req: UpdateReqType<typeof schemas.examEnvironmentGetExamChallenge>,
   reply: FastifyReply
 ) {
   const logger = this.log.child({ req });
-  const { challengeId } = req.params;
+  const { challengeId, examId } = req.query;
 
-  logger.info({ challengeId });
+  logger.info({ challengeId, examId });
 
   const maybeData = await mapErr(
     this.prisma.examEnvironmentChallenge.findMany({
       where: {
-        challengeId
+        challengeId: challengeId ?? undefined,
+        examId: examId ?? undefined
```

Comment:
I made a `GET /exam-environment/exam-challenge` request here with no query params and it returned the seeded challenge from the database. Is that intended? Agent says...

"When both conditions are `undefined`, Prisma ignores those fields, effectively making it an empty where clause:

```
this.prisma.examEnvironmentChallenge.findMany({
  where: {}
})
```

Which returns all records"

Looks like the client shouldn't

Reply:
This was unintentional. I considered making an/the endpoint handle returning all, but that could be abused, as this is public. So, I will definitely address this.